__device-tests__/helpers/utils.js

```diff
@@ -118,10 +126,92 @@ const stopDriver = async ( driver: wd.PromiseChainWebdriver ) => {
 	}
 };
 
+// attempts to type a string to a given element, need for this stems from
+// https://github.com/appium/appium/issues/12285#issuecomment-471872239
+// https://github.com/facebook/WebDriverAgent/issues/1084
+const typeString = async ( driver: wd.PromiseChainWebdriver, element: wd.PromiseChainWebdriver.Element, str: string, clear: boolean = false ) => {
+	if ( clear ) {
+		await element.clear();
+	}
+
+	if ( isAndroid() ) {
+		if ( str in strToKeycode ) {
+			return await driver.pressKeycode( strToKeycode[ str ] );
```

Comment:
We can probably remove this part since it will be handled by the paragraph logic (since `'\n'.split('\n')` gives us `["", ""]`)

Reply:
Hmm I think it wouldn't be handled in the paragraph logic for the case the string is `'\u0008'`. The split would cause the paragraphs length to be equal to 1 and then try to execute it like regular text instead of pressing the keycode. So this statement is just for the case is a single command, alternatively if the if statement is changed to `<=` it might handle it though right? 🤔

Reply:
Oh right, I missed the `if ( paragraphs.length > 1 ) {`!
I wonder why it's there though, is `paragraphs.length` the logic inside the loop still makes sense right?

Reply:
> if the if statement is changed to <= it might handle it though right? 🤔

You mean `> 0` ? In that case yes :)

Reply:
> You mean > 0 ? In that case yes :)

🤦🏽‍♂️ 😅 Yes that!

Reply:
Actually, we can't remove this part because of the Unicode characters, like the one used for backspace '\u0008', splitting on this will return an empty string and not send the keyevent.

Reply:
I'm not sure I understand, `'\u0008'.split( '\n' )` returns `[ '\u0008' ]`.
Then in your loop, `paragraph in strToKeycode` will be true and it will send the right keystroke.

That said it works fine at the moment and it's not bad per se so I don't mind too much if we keep it that way